File: ProblemSolve/474.cs
/******************
 * One and Zeroes *
 ******************/

public class Solution {
    public int count(string str, char ch){
        int result = 0;
        
        for(int i=0; i<str.Length; ++i){
            if(str[i] == ch){
                result++;
            }
        }
        
        return result;
    }
    
    public int FindMaxForm(string[] strs, int m, int n) {
        int[,] dp = new int[m+1, n+1];
        
        int zeroCount = 0, oneCount = 0;
        
        foreach(var str in strs){
            zeroCount = count(str, '0');
            oneCount = str.Length - zeroCount;
            
            for(int i=m; i >= zeroCount; --i){
                for(int j=n; j >= oneCount; --j){
                    dp[i,j] = Math.Max(dp[i,j], dp[i-zeroCount, j-oneCount] + 1);
                }
            }
        }
        
        return dp[m,n];
    }
}

File: ProblemSolve/474.cpp
/******************
 * One and Zeroes *
 ******************/

class Solution {
public:
    int findMaxForm(vector<string>& strs, int m, int n) {
        vector<vector<int>> dp(m + 1, vector<int>(n+1,0));
        
        int zeroCount = 0, oneCount = 0;
        
        for(int i=0; i<strs.size(); ++i){
            int zeroCount = count(strs[i].begin(), strs[i].end(), '0');
            int oneCount = strs[i].size() - zeroCount;
            
            for(int j = m; j >= zeroCount; --j){
                for(int k = n; k >= oneCount; --k){
                    dp[j][k] = max(dp[j][k], dp[j - zeroCount][k - oneCount] + 1);
                }
            }
        }
        
        return dp[m][n];
    }
};